optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/common/BavetNodeBuildPolicy.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.bavet.common;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.optaplanner.core.impl.score.stream.bavet.BavetConstraintSession;
+
+public class BavetNodeBuildPolicy<Solution_> {
+
+    private final BavetConstraintSession session;
+
+    private int nodeOrderMaximum = 0;
```

Comment:
This variable is never used in this class. It has a getter and a setter, but the class itself doesn't use it anywhere.
That suggests to me that, even though it may still be useful, it perhaps doesn't belong here?

Reply:
After building the nodes, we need the nodeOrderMaximum for performance (list capacity). The BuildPolicy is there during building all nodes, so it's easy to just check every nodeOrder number to figure out the largest.
That being said, we could also just look at the constraint list - after building the nodes - and grab the largest there. Although Constraint's don't know their ScoringStream as a parent...